File: ShittyText.java
import java.util.ArrayList;

public class ShittyText {

    private static ArrayList<WordFollowers> words = new ArrayList<WordFollowers>();

    private static void addWord(String word, String follower) {
        for (WordFollowers existing : words) {
            if (existing.word.equals(word)) {
                existing.addFollower(follower);
                return;
            }
        }
        words.add(new WordFollowers(word, follower));
    }

    private static WordFollowers findWF(String word) {
        for (int i = 0; i < words.size(); i++) {
            WordFollowers wf = words.get(i);
            if (wf.word.equals(word)) {
                return wf;
            }
        }
        return null;
    }

    private static String generateText(int wordCap) {
        // Get random starting word
        int rand = (int) Math.floor(Math.random() * words.size());
        WordFollowers wf = words.get(rand);
        String word = wf.word;

        String sentence = word;
        while (wordCap > 0) {
            String newFollower = wf.getWRandomFollower();
            wf = findWF(newFollower);
            if (wf == null) {
                break;
            }
            sentence += " " + newFollower;
            wordCap--;
        }
        return sentence;
    }

    public static void main(String[] args) {
        for (int i = 0; i < args.length-1; i++) {
            addWord(args[i], args[i+1]);
        }

        System.out.println(generateText(10));
    }

}

File: ShittyText.py
import wikipedia as w, random, re

class WordFollowers():
    def __init__(self, word, follower):
        self.word = word
        self.followers = [follower]

    def addFollower(self, follower):
        self.followers.append(follower)

    def getFollower(self):
        return random.choice(self.followers)

def shittytext(searchTerm):
    words = []
    articles = w.search(searchTerm)
    if len(articles) == 0:
        return "Search term not found"
    
    plaintext = re.sub('[^a-zA-Z ]+', '', w.summary(articles[0]))
    data = plaintext.lower().split(" ")

    def addWordFollower(word, follower):
        for j in words:
            if j.word == word:
                j.addFollower(follower)
                return
        words.append(WordFollowers(word, follower))

    for i in range(0, len(data)-1):
        addWordFollower(data[i], data[i+1])

    def findFollower(follower):
        for i in words:
            if i.word == follower:
                return i
        return None

    wordCap = 50
    word = random.choice(words)
    sentence = [word.word]
    while (wordCap > 0):
        follower = word.getFollower()
        sentence.append(follower)
        word = findFollower(follower)
        if (word == None):
            word = random.choice(words)
        wordCap -= 1

    return ' '.join(sentence)
